### PR TITLE
Set up pre-commit hooks and workflows to check code style

### PR DESCRIPTION
**Pull request by aws-tianquaw**
*Issue #, if available:*
Currently the code base doesn't have any code stile check machnism.

*Description of changes:*
- Add pre-commit hooks to run code style formatters including black, isort, autoflake
- Run formatters on all existing files in this code base
- Add a git action workflow to check code styles whenever a new pull request is created and whenever something is pushed

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -24,6 +24,23 @@ Run the following to invoke those tests:
 pytest --local-image-id REPLACE_ME_WITH_IMAGE_ID
 ```
 
+## Code Style
+
+Install pre-commit to run code style checks before each commit:
+
+```shell
+pip install pre-commit
```
Comment (claytonparnell) [inline DEVELOPMENT.md:32]: given we have instructed to conda install the environment.yml which includes pre-commit, we don't need to additionally pip install the package.

**Review by claytonparnell — CHANGES_REQUESTED** [review]
We should run all the unit tests (tests/test_main.py etc.) to make sure none of the functionality is impacted. And probably run a build/test/push as well.

**Comment by balajisankar15** [comment]
Once this is merged, can you also rebase 2.0.0-beta with these changes?

**Comment by aws-tianquaw** [comment]
> We should run all the unit tests (tests/test_main.py etc.) to make sure none of the functionality is impacted. And probably run a build/test/push as well.

Executed the local build/test and got the following results:
```
=========================== short test summary info ============================
FAILED tests/unit/sagemaker/serve/builder/test_transformers_builder.py::TestTransformersBuilder::test_build_deploy_for_trans
formers_local_container_and_remote_container
FAILED tests/unit/sagemaker/serve/model_server/triton/test_server.py::TritonServerTests::test_start_invoke_destroy_local_tri
ton_server_cpu
FAILED tests/unit/sagemaker/serve/model_server/triton/test_server.py::TritonServerTests::test_start_invoke_destroy_local_tri
ton_server_gpu
= 3 failed, 7668 passed, 3778 skipped, 2 deselected, 3 xpassed, 3308 warnings in 1619.18s (0:26:59)
```
The failed tests exactly match what is documented in https://github.com/aws/sagemaker-distribution/issues/316, so we can conclude that this PR doesn't introduce any new test failure.

**Comment by aws-tianquaw** [comment]
> Once this is merged, can you also rebase 2.0.0-beta with these changes?

Sure I'll make sure to do it once this PR is merged

**Review by claytonparnell — APPROVED** [review]

**Review by balajisankar15 — APPROVED** [review]

**Pull request merged**